common/test/acceptance/tests/video/test_video_events.py

```diff
@@ -47,33 +94,193 @@ def is_video_event(event):
+
+
+@ddt.ddt
+class VideoBumperEventsTest(VideoEventsTestMixin):
+    """ Test bumper video event emission """
+
+    # helper methods
+    def seek_video_and_skip(self):
+        """
+        Wait 5 seconds and press "skip" button.
+        """
+        self.video.wait_for_position('0:05')
+        self.video.click_player_button('skip_bumper')
+
+    def seek_video_and_dismiss(self):
+        """
+        Wait 5 seconds and press "do not show again" button.
+        """
+        self.video.wait_for_position('0:05')
+        self.video.click_player_button('do_not_show_again')
+
+    def wait_for_state(self, state='finished'):
+        """
+        Wait until video will be in given state.
+
+        Finished state means that video is played to the end.
+        """
+        self.video.wait_for_state(state)
+
+    def add_bumper(self):
+        """
+        Add video bumper to the course.
+        """
+        additional_data = {
+            u'video_bumper': {
+                u'value': {
+                    "transcripts": {},
+                    "video_id": "edx_video_id"
+                }
+            }
+        }
+        self.course_fixture.add_advanced_settings(additional_data)
+
+    @ddt.data(
+        ('edx.video.bumper.skipped', seek_video_and_skip),
+        ('edx.video.bumper.dismissed', seek_video_and_dismiss),
+        ('edx.video.bumper.stopped', wait_for_state)
+    )
+    @ddt.unpack
+    def test_video_control_events(self, event_type, action):
+        """
+        Scenario: Video component is rendered in the LMS in Youtube mode without HTML5 sources
+        Given the course has a Video component in "Youtube" mode
+        And I play the video
+        And I watch 5 seconds of it
+        And I pause the video
+        Then a "load_video" event is emitted
+        And a "play_video" event is emitted
+        And a "pause_video" event is emitted
+        """
+
+        def is_video_event(event):
+            """Filter out anything other than the video events of interest"""
+            return event['event_type'] in (
+                'edx.video.bumper.loaded',
+                'edx.video.bumper.played',
+                'edx.video.bumper.skipped',
+                'edx.video.bumper.dismissed',
+                'edx.video.bumper.stopped',
+                'load_video',
+                'play_video',
+                'pause_video'
+            ) and self.video.state != 'buffering'
+
+        captured_events = []
+        self.add_bumper()
+        with self.capture_events(is_video_event, number_of_matches=5, captured_events=captured_events):
+            self.navigate_to_video_no_render()
+            self.video.click_on_poster()
+            self.video.wait_for_video_bumper_render()
+            sources, duration = self.video.sources[0], self.video.duration
+            action(self)
+
+        # Filter subsequent events that appear due to bufferisation: edx.video.bumper.played
```

Comment:
@polesye wrote in reply:  "No, it is another one issue. In our case, the video is playing well, but due to buffering it emits more than one `edx.video.bumper.played` event and our acceptance tests fail. Events list looks so:
[u'edx.video.bumper.loaded', u'edx.video.bumper.played', u'edx.video.bumper.played',  u'edx.video.bumper.stopped', u'load_video', u'play_video’].

When we started emitting `waited` event that occurs when the video stops because it needs to buffer the next frame, we have got the following list of events that proves our assumpations:
[u'edx.video.bumper.loaded', u'edx.video.bumper.waited', u'edx.video.bumper.waited', u'edx.video.bumper.played', u'edx.video.bumper.waited', u'edx.video.bumper.played',  u'edx.video.bumper.stopped', u'load_video', u'play_video’].
To fix the multiple occurrences of edx.video.bumper.played,  the video player needs to be refactored, but it should be a part of another PR ."